indico/modules/events/contributions/operations.py

```diff
@@ -181,3 +182,22 @@ def create_contribution_from_abstract(abstract, contrib_session=None):
                                        'session': contrib_session,
                                        'person_link_data': {link: True for link in contrib_person_links}},
                                custom_fields_data=custom_fields_data)
+
+
+def clone_contribution(event, contribution, preserve_session):
+    shared_data = {
+        'event_persons': {'person_map': {person: person for person in event.persons}},
+        'sessions': {
+            'session_map': {},
+            'session_block_map': {}
+        },
+        'contribution_fields': {'contrib_field_map': {field: field for field in event.contribution_fields}},
+        'contribution_types': {'contrib_type_map': {type_: type_ for type_ in event.contribution_types}}
+    }
+
+    if preserve_session:
+        shared_data['sessions']['session_map'][contribution.session] = contribution.session
```

Comment:
You could always populate this above; no harm in having it even when the session is not used.